Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -361,6 +380,154 @@ public sealed class Server : ServerBase, IDisposable {
             return item;
         }
 
+        public override async Task<SignatureHelp> SignatureHelp(TextDocumentPositionParams @params) {
```

Comment:
This method is 100% sync. Is it by design?

Reply:
It is right now, but in the future it may not be. This function is public stable API, so keeping it a Task even if it doesn't matter yet is fine, and historically we've missed enough `Task.FromResult(null)` calls to cause crashes that appear to come from other places in code, so leaving it `async` is also safer. I'm not concerned about the performance implications of this particular call.